SEP-0012.md

```diff
@@ -0,0 +1,495 @@
+        """
+        The mapping between the array dimensions and pixel dimensions.
+
+        This is an iterable of ``(array_dimension, pixel_dimension)`` pairs
+        of length equal to the number of pixel dimensions in the extra coords.
+        """
+
+    @abc.abstractmethod
+    @property
+    def wcs(self) -> BaseHighLevelWCS:
+        """
+        A WCS object representing the world coordinates described by this ``ExtraCoords``.
+
+        .. note::
+            This WCS object does not map to the pixel dimensions of the array
+            associated with the `.NDCube` object. It has the number of
+            pixel dimensions equal to the number of inputs to the transforms to
+            get the world coordinates (normally equal to the number of world
+            coordinates). Therefore using this WCS directly might lead to some
+            confusing results.
+
+        """
+
+    @abc.abstractmethod
+    def __getitem__(self, item: Union[str, int, slice, Iterable[Union[str, int, slice]]]) -> "ExtraCoordsABC":
+        """
+        ExtraCoords can be sliced with either a string, or a numpy like slice.
+
+        When sliced with a string it should return a new ExtraCoords object
+        with only those coordinates with the given names. When sliced with a
+        numpy array like slice it should return a new ExtraCoords with the
+        slice applied. Supporting step is not required and "fancy indexing" is
+        not supported.
+        """
+```
+
+## Global Coords
+
+The final part of the `NDCube` API is a class to provide structured metadata for coordinate values which are applicable to the whole cube.
+One possible use (not mandated by this SEP) is to have any coordinates dropped by a slicing operation on a higher dimensionality cube be tracked here.
+This class is only a slight extension of the API for an immutable dictionary, to enable the tracking of the physical type of the coordinate as well as it's name and value.
+The class behaves like a mapping between coordinate name and coordinate object for all read operations.
+
+
+```python
+
+from collections.abc import Mapping
+
+
+class GlobalCoordsABC(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+
+    This class acts as a mapping between coordinate name and the coordinate object.
+    In addition to this a physical type is stored for each coordinate name.
+    A concrete implementation of this class must furfill the `Mapping` ABC,
+    including methods such as `__iter__` and `__len__`.
+
+    Parameters
+    ----------
+    ndcube : `.NDCube`, optional
+        The parent ndcube for this object. Used to extract global coordinates
+        from the wcs and extra coords of the ndcube. If not specified only
+        coordinates explicitly added will be shown.
+    """
+    def __init__(self, ndcube: Optional[NDCubeABC] = None):
+        pass
+
+    def add(self,
```

Comment:
Cross posting from https://github.com/sunpy/ndcube/issues/348

@DanRyanIrish suggests this be renamed `add_coordinate` to match the `ExtraCoords` class, I would also in tandem rename `remove` to `remove_coordinate`.

Reply:
I am in favour of this proposal :+1:

Reply:
Me too, looks nice 👍

Reply:
Alternatively, `ExtraCoords.add_coordinate` can be changed to `.add`.  I think that's nice and clean.

Reply:
https://github.com/sunpy/sunpy-SEP/pull/51/files#r533333588